CRM/Iats/iATSServiceRequest.php

```diff
@@ -272,8 +272,8 @@ public function result($response, $log = TRUE) {
       case 'process':
         if (!empty($response->PROCESSRESULT)) {
           $processresult = $response->PROCESSRESULT;
-          $result['auth_result'] = trim(current($processresult->AUTHORIZATIONRESULT));
-          $result['remote_id'] = current($processresult->TRANSACTIONID);
+          $result['auth_result'] = trim(((array) $processresult->AUTHORIZATIONRESULT)[0] ?? '');
```

Comment:
$result['auth_result'] = is_array($processresult->AUTHORIZATIONRESULT) ? trim(current($processresult->AUTHORIZATIONRESULT)) : '';

Reply:
I confess I'm scared of weird side effects every time I look at this. Is my alternative above reasonable? It feels a little more robust at handling unexpected results.

Reply:
Your alternative looks "better", but it might give a different answer than the current handling, e.g. if AUTHORIZATIONRESULT was a string instead of an array.

Reply:
So `$processresult->AUTHORIZATIONRESULT` is a simplexmlelement, and is_array() returns false on it. So that won't work as-is.

If for some reason instead it comes in as a string sometimes, then currently you get a fatal error because `current` doesn't like that, so yes you're right it would be a change since with the PR you'd get the string value. What do we want it to do if it is a string? Do we want it to stay as an error?

Reply:
Okay, I'm convinced, I like your code.